### PR TITLE
SSL Support

### PR DESCRIPTION
**Pull request by ferdinandpolpol**
This branch adds ssl certificate to openosp if the keys are provided. Solves Issue #11

### CODE REVIEW

**Comment by ferdinandpolpol** [comment]
@countable could you review?

File: conf/nginx/conf.d/prod.conf [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+}
+
+server {
+        listen 443 ssl;
+        
+        server_name openosp.ca;
+        
+        ssl_certificate      /ssl/certww.pem;
+        ssl_certificate_key  /ssl/privkey.pem;
```
Comment (countable) [inline conf/nginx/conf.d/prod.conf]: @ferdinandpolpol we still want everything (oscar) to work on https, so don't we need to proxy it to port 80, or repeat/include the same location code?

File: conf/nginx/conf.d/prod.conf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                proxy_set_header Connection "upgrade";
+
+                sub_filter_types text/css;
+
+                sub_filter 'OSCAR McMaster Trunk' '<div id="login-title">${LOGIN_TITLE}</div>';
+                sub_filter 'images/OSCAR-LOGO.gif' '/static/images/OSCAR-LOGO.png';
+                sub_filter 'img src' 'img id="login-logo" src';
+                sub_filter 'build.dateTime' '${BUILD_DATE}';
+                sub_filter 'build.JOB_NAME' '${BUILD_NAME}';
+                sub_filter 'build.BUIL_DNUMBER' '${BUILD_NUMBER}';
```
Comment (countable) [inline conf/nginx/conf.d/prod.conf]: Seems like a typo BUIL_DNUMBER

File: oscar-login.env [hunk 3]
```diff
@@ -1,5 +1,5 @@
-LOGIN_TEXT=<div  id="login-text"><p>This to the OpenOSP Free Oscar Training Instance.</p><p>If you want a separate one of your own, fill out this <a href="https://docs.google.com/forms/d/1U5zOtFWE2hb6HgG9NeE9x2Ol3p9L28gUJNkSpEpksHI/edit" target="_blank">form</a>.</p><p>As a training instance, DO NOT USE REAL PATIENT DATA here. The data is not backed up, and this service is offered with out warranty of any kind. It is intended strictly for testing and training new versions of Oscar.</p></div>
```
Comment (countable) [inline oscar-login.env]: @ferdinandpolpol we should actually keep all this other stuff, too. Sorry

**Comment by ferdinandpolpol** [comment]
@countable Update Readme on how to generate SSL, update prod.conf to redirect http to https

**Pull request merged**